ci/scripts/pr_list_database.py

```diff
@@ -0,0 +1,94 @@
+#!/usr/bin/env python3
+
+import sys
+from pathlib import Path
+import sqlite3
+from argparse import ArgumentParser, ArgumentDefaultsHelpFormatter
+
+
+def sql_connection(filename):
+    try:
+        con = sqlite3.connect(Path(filename))
+        return con
+    except sqlite3.Error:
+        print(sqlite3.Error)
+        sys.exit(-1)
+
+
+def sql_table(obj):
+    obj.execute("CREATE TABLE processing(pr integer PRIMARY KEY, state text, status text)")
+
+
+def sql_insert(obj, entities):
+    obj.execute('INSERT INTO processing(pr, state, status ) VALUES(?, ?, ?)', entities)
+
+
+def sql_update(obj, pr, state, status):
+    obj.execute(f'UPDATE processing SET state = "{state}", status = "{status}" WHERE pr = {pr}')
+
+
+def sql_fetch(obj):
+    obj.execute('SELECT * FROM processing')
+    rows = obj.fetchall()
+    return rows
+
+
+def sql_remove(obj, pr):
+    obj.execute(f'DELETE FROM processing WHERE pr = {pr}').rowcount
+
+
+def input_args():
+
+    description = """Arguemtns for creating and updating db file for pr states
+    """
+
+    parser = ArgumentParser(description=description,
+                            formatter_class=ArgumentDefaultsHelpFormatter)
+
+    parser.add_argument('sbfile', help='file of sqlite3 for pr list database ', type=str)
+    parser.add_argument('--create', help='create sqlite file for pr list status', action='store_true', required=False)
+    parser.add_argument('--add_pr', nargs=1, metavar='PR', help='add new pr to list (defults to: Open,Ready)', required=False)
+    parser.add_argument('--remove_pr', nargs=1, metavar='PR', help='removes pr from list', required=False)
+    parser.add_argument('--update_pr', nargs=3, metavar=('pr', 'state', 'status'), help='updates state and status of a given pr', required=False)
+    parser.add_argument('--display', help='output pr table', action='store_true', required=False)
```

Comment:
Given these are probably mutually exclusive (or, at least, we shouldn't do multiple at once), I'm not sure this is the best method to capture the options. Probably want a single positional argument with a limited list of choices: `['create', 'add', 'update', 'remove', 'display']` with subparsers to handle add the additional arguments needed for each as setup_expt.py uses.

Reply:
Nice, makes sense to make it positional   Yes lets make this an update in the next PR, thanks.